toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -658,25 +664,139 @@ func convertPackageVersionToTdnfArg(pkgVer *pkgjson.PackageVer) (tdnfArg string)
 	// To avoid significant overhead we only download the latest version of a package
 	// for ">" and ">=" constraints (ie remove constraints).
 	switch pkgVer.Condition {
-	case "<=", "<", "=":
+	case "":
+	case "=":
+		tdnfArg = fmt.Sprintf("%s-%s", pkgVer.Name, pkgVer.Version)
+	case "<=", "<":
 		tdnfArg = fmt.Sprintf("%s %s %s", pkgVer.Name, pkgVer.Condition, pkgVer.Version)
-	case ">", ">=", "":
-		if pkgVer.Condition != "" {
-			logger.Log.Warnf("Discarding '%s' version constraint for: %v", pkgVer.Condition, pkgVer)
-		}
+	case ">", ">=":
+		logger.Log.Warnf("Discarding '%s' version constraint for: %v", pkgVer.Condition, pkgVer)
 	default:
 		logger.Log.Errorf("Unsupported version constraint: %s", pkgVer.Condition)
 	}
 
 	return
 }
 
-// selectCorrectCacheRepoID determines which cache repo we are using, the normal one, or the pre-mounted one for use with
-// containers.
-func selectCorrectCacheRepoID() string {
-	if buildpipeline.IsRegularBuild() {
-		return fetcherRepoID
-	} else {
-		return cacheRepoID
+// GetEnabledRepos returns the repo flags that the cloner is allowed to use for its queries.
+func (r *RpmRepoCloner) GetEnabledRepos() uint64 {
+	return r.reposFlags
+}
+
+// SetEnabledRepos tells the cloner which repos it is allowed to use for its queries.
+func (r *RpmRepoCloner) SetEnabledRepos(reposFlags uint64) {
+	r.reposFlags = reposFlags
+	r.reposArgsList = [][]string{}
+	previousReposList := []string{fmt.Sprintf("--disablerepo=%s", repoIDAll)}
+
+	defer func() {
+		logger.Log.Debugf("Enabled repos: %v.", r.reposArgsList)
+	}()
+
+	// Do NOT change the order of the following 'if' statements!
+	// The order is critical as we want to gradually enable repositories in the following order:
+	// 1. Toolchain.
+	// 2. Locally-built packages.
+	// 3. Local cache of packages downloaded from external sources.
+	// 4. Upstream repositories requiring network access.
+	if RepoFlagToolchain&reposFlags != 0 {
+		previousReposList = append(previousReposList, fmt.Sprintf("--enablerepo=%s", repoIDToolchain))
+		r.reposArgsList = append(r.reposArgsList, previousReposList)
+	}
+
+	if RepoFlagLocalBuilds&reposFlags != 0 {
+		previousReposList = append(previousReposList, fmt.Sprintf("--enablerepo=%s", repoIDBuilt))
+		r.reposArgsList = append(r.reposArgsList, previousReposList)
+	}
+
+	if RepoFlagDownloadedCache&reposFlags != 0 {
+		previousReposList = append(previousReposList, fmt.Sprintf("--enablerepo=%s", r.repoIDCache))
+		r.reposArgsList = append(r.reposArgsList, previousReposList)
+	}
+
+	// Options past this point are only valid if upstream repos are enabled.
+	if RepoFlagUpstream&reposFlags == 0 {
+		return
+	}
+
```

Comment:
Do we want to preferentially enable the user's selected repos before resorting to `"*"`?

Reply:
Per offline discussion, we rely on passing tooling related repos alongside custom repos and it is non-trivial to differentiate them.